flake.nix

```
@@ -49,12 +49,22 @@
                   })).config;
 
                 moduleDeclarationFile =
-                  (builtins.unsafeGetAttrPos "modules" args).file;
+                  let
+                    # Even though `modules` is a mandatory argument for `nixosSystem`, it doesn't
+                    # mean that the evaluator keeps track of its position. If there are too many levels
+                    # of indirection, the position gets lost at some point and `EvalState::mkPos`
+                    # returns `null` in `libexpr`.
+                    #
+                    # Such an issue was found in https://github.com/NixOS/nixpkgs/commit/e14c24593420bb9057e7f38b40d17137eaeff9dd#commitcomment-53645936
+                    # / https://github.com/divnix/digga/issues/87
+                    intermediatePos = builtins.unsafeGetAttrPos "modules" args;
```

I don't expect this to be relevant to the reader. You could put the link in the commit message instead.

---

Sorry, I referenced one line too many here. My comment only applies to the inline comment

```suggestion
                    intermediatePos = builtins.unsafeGetAttrPos "modules" args;
```